openedx/custom/db/fields.py

```diff
@@ -0,0 +1,79 @@
+from django.core import exceptions
+from django.utils.text import capfirst
+from multiselectfield import MultiSelectField
+from multiselectfield.db.fields import MSFList
+from multiselectfield.utils import string_type, get_max_length
+
+from openedx.custom.forms.fields import MultiSelectWithOtherFormField
+
+
+class MultiSelectWithOtherField(MultiSelectField):
+
+    def __init__(self, other_max_length=None, *args, **kwargs):
+        self.other_max_length = other_max_length
+
+        if kwargs.get('max_length') is None and self.other_max_length is not None:
+            choice_max_length = get_max_length(kwargs['choices'], kwargs.get('max_length'))
+            kwargs['max_length'] = choice_max_length + self.other_max_length
+
+        if type(kwargs['choices']) == tuple:
+            kwargs['choices'] = kwargs.pop('choices') + (('other', 'Other'),)
+
+        super(MultiSelectWithOtherField, self).__init__(*args, **kwargs)
+
+        self.error_messages.update({
+            'invalid_char': 'value %s contains invalid character `|`'
```

Comment:
Little late but it would be great if you use magic constant for OTHER_SEPARATOR `|`